src/docs/stan-reference/examples.tex

```diff
@@ -8046,6 +8046,154 @@ \section{Circular Representations of Days and Years}
 weekends, or with data normalization back to natural scales for
 daylight savings time.
 
+\chapter{Solving Algebraic Equations}\label{algebra-solver.chapter}
+
+\noindent
+Stan provides a built-in mechanism for specifying and solving systems
+of algebraic equations. The function signatures for Stan's algebraic solver
+can be found in \refchapter{functions-algebraic-solver}.
+%
+Solving any system of algebraic equations can be translated into a root-finding
+problem, that is, given a function $f(x)$, we wish to find $x$ such that
+$f(x) = 0$.
+
+\section{Example: System of Nonlinear Algebraic Equations}
+
+For systems of linear algebraic equations, we recommend solving the system
+using an inverse matrix. The algebraic solver becomes handy when we want
+to solve nonlinear equations.
+%
+As an illustrative example, we consider a nonlinear system of two equations
+with two unknowns. Our goal is to simultaneously solve all equations for
+$x_1$ and $x_2$, such that the vector function $z$ goes to 0. 
+%
+\begin{eqnarray}\label{algebra.equation}
+  \begin{aligned}
+  z_1 &= x_1 - \theta_1 \\
+  z_2 &= x_1 x_2 + \theta_2 \\
+  \end{aligned}
+\end{eqnarray}
+%
+
+\section{Coding an Algebraic System}
+
+A system of algebraic equations is coded directly in Stan as a function with a
+strictly specified signature. For example, the nonlinear system given by
+\refequation{algebra.equation} can be coded using the following function
+in Stan (see \refchapter{functions-programming} for
+more information on coding user-defined functions).
+%
+\begin{stancode}
+real[] system(real[] x,       // unknowns
+               real[] theta,  // parameters
+               real[] dat_r,    // data (real)
+               real[] dat_i) {  // data (integer)
+  real z[2];
+  z[1] = x[1] - theta[1];
+  z[2] = x[1] * x[2] - theta[2];
+  return z;
+}
+\end{stancode}
+%
+The function takes the unknowns we wish to solve for in \code{x} (a real array),
+the system parameters in \code{theta} (a real array), the real data in
+\code{dat\_r} (a real array) and the integer data in \code{dat\_i} (an integer array). The
+system function returns the value of the function, for which we want to compute the roots. 
+Our example does not use real or integer data. Nevertheless, these
+unused arguments must be included in the system function with exactly the
+signature above.
+
+\subsubsection{Strict Signature}  
+%
+The function defining the system must have exactly these argument types and
+return type. This may require passing in zero-length arrays for data or parameter
+if the system does not involve data or parameters.
+
+\section{Calling the Algebraic Solver}
+%
+Let's suppose $\theta = \{3, 6\}$. To call the algebraic solver, we need to
+provide an initial guess. A good guess can significantly speed up the solver and
+even determine whether the solver converges or not.
+%
+The following code returns the solution to our nonlinear algebraic system:
+%
+\begin{stancode}
+transformed data {
```

Comment:
I got mixed up with the ODE integrators. Yes, because the algebraic solver is based on Eigen's dogleg, the returned object, initial guess and parameters are vectors. The real and integer data are vectors.